src/lib/zf/cplane.c

```diff
@@ -231,10 +241,16 @@ __zf_cplane_get_path(struct zf_stack* st, struct zf_path* path,
     /* In theory, we shoud break out if we see FLAG_ARP_FAILED popping up
      * in a non-first loop.  But if we see ARP_FAILED from the start, we
      * should give a chance to re-resolve.  For now we always loop. */
-    if( (data.flags & CICP_FWD_DATA_FLAG_ARP_VALID) || iter == 0 )
+    if( (data.flags & CICP_FWD_DATA_FLAG_ARP_VALID) || (time_limit == 0) )
+      break;
+
+    gettimeofday(&now, NULL);
+    time_used = ((now.tv_sec - start.tv_sec) * 1000000) +
+                (now.tv_usec - start.tv_usec);
+
+    if (time_used >= time_limit)
       break;
 
-    iter--;
     usleep(1);
```

Comment:
This would make the loop potentially very aggressive in polling the CP server, which may cause issues with other processes attempting to do the same. This looks to be limited to memory accesses though, since there are no blocking calls or locks I can see.

Reply:
I've opted to keep the sleep. This adds 1us on each iteration, which takes 22us so the additional time is small compared to a potentially aggressive polling.